dapp/src/components/shared/Form/FormTokenBalanceInput.tsx

```diff
@@ -28,14 +29,28 @@ export function FormTokenBalanceInput({
     }
   }, [defaultValue, setAmount])
 
+  const handleChange = useCallback<ChangeEventHandler>(
+    (event) => {
+      field?.onChange(event)
+
+      if (!meta.error) {
+        return
+      }
+
+      helpers.setError(undefined)
+    },
+    [field, meta, helpers],
+  )
+
```

Comment:
Non-blocking comment

I wonder why not just do it in `setAmount`. Note that in this function we also set `touched` to `true`. I would suggest keeping such changes together to maintain some consistency. So that later we don't wonder whether further changes should be added to `setAmount` or `handleChange`. What do you think about it?

https://github.com/thesis/acre/blob/dc156f5a7f02142c1f80627267d14a26e5c99b30/dapp/src/components/shared/Form/FormTokenBalanceInput.tsx#L19-L20

We could then simplify the code to

```
  const setAmount = useCallback(
    (value?: bigint) => {
      if (!meta.touched) logPromiseFailure(helpers.setTouched(true))
      if (meta.error) helpers.setError(undefined)

      logPromiseFailure(helpers.setValue(value))
    },
    [helpers, meta.error, meta.touched],
  )
```

Reply:
That's one of the solutions. Have to admit your suggestion is cleaner. Thanks
Ref commit: ea855814